silo/views.py

```diff
@@ -936,9 +936,13 @@ def getJSON(request):
 def toggle_silo_publicity(request):
     silo_id = request.GET.get('silo_id', None)
     silo = Silo.objects.get(pk=silo_id)
-    silo.public = not silo.public
-    silo.save()
-    return HttpResponse("Your change has been saved")
+
+    if silo.owner == request.user:
+        silo.public = not silo.public
+        silo.save()
+        return HttpResponse(status=200)
+    else:
+        return HttpResponse(status=401)
```

Comment:
You can check the following URL to understand better the status code and which one we should use. It helps me as well :smile: 
https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html